### PR TITLE
chore: use same version of @xstate/react across amplify-ui

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Use same version of @xstate/react across amplify-ui #4027

#### Issue #, if available

Fixes: https://github.com/aws-amplify/amplify-ui/issues/4025

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by reesscot** [comment]
What's currently using 3.0.1?

**Comment by sreeramsama** [comment]
> What's currently using 3.0.1?

`react-core`

This PR addresses https://github.com/aws-amplify/amplify-ui/issues/4025

File: examples/react-native/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "@aws-amplify/ui-react-native": "*",
     "@react-native-async-storage/async-storage": "^1.17.5",
     "@react-native-community/netinfo": "^8.3.0",
-    "@xstate/react": "3.0.0",
+    "@xstate/react": "3.0.1",
```
Comment (calebpollman) [inline examples/react-native/package.json:22]: Why do we need `xstate/react` in this package (or _@aws-amplify/ui-react_)? Shouldn't the dep in _@aws-amplify/ui-react-core_ cover it?

Comment (ioanabrooks) [reply examples/react-native/package.json:22]: +1

Comment (sreeramsama) [reply examples/react-native/package.json]: Yeah, true.
We don't need to have it again.
Updated PR.

**Comment by kachkaev** [comment]
Can `@xstate/react` be updated to 3.2.2 given the opportunity?

**Comment by reesscot** [comment]
@kachkaev I think we would want to address that in a follow up PR since that would require more testing and this will fix the mismatched versions issue. Mind opening a separate feature request to bump the xstate version to the latest?

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**